docs/about.md

```diff
@@ -0,0 +1,49 @@
+## Mailing list
+
+OpenBLAS users: <https://groups.google.com/forum/#!forum/openblas-users>
+
+OpenBLAS developers: <https://groups.google.com/forum/#!forum/openblas-dev>
+
+## Donations
```

Comment:
This is all quite old, but that can be improved in a follow-up I think. The Google Docs link still works; last entries are from 2016.

Reply:
xianyi created a workspace on slack after that, but given his somewhat elusive nature not much happened there AFAIK (and it would be by invitation only anyway). So might as well keep the google docs link for now (though perhaps nowadays the github `Discussions` tab is more practical ?).

Reply:
https://github.com/OpenMathLib/OpenBLAS/discussions/ indeed seems worth linking here